perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/scheduling/PropagationMaintainer.java

```diff
@@ -107,6 +170,8 @@ public void run() {
 			return;
 		}
 
+		this.createIndependentTaskSchedules();
```

Comment:
You should make sure, that repetitively scheduled tasks are not re-sheduled in below methods. I think, that all Tasks are in the shared pool and other threads work with them, so you should eg. skip Tasks with ``scheduleExpr != NULL`` in their processing.

Reply:
I disagree with that too. I tend to think that tasks should be scheduled at least once in two days (as a safety in case the periodic scheduling fails for whatever reason), should be run when new data appears, and as above, should be subject to the same reschedule rules in case of failure.